technical/hashmap/c++/hashmap.h

```diff
@@ -0,0 +1,131 @@
+#include <vector>
+#include <string>
+
+static const size_t MIN_CAPACITY = 4;
+static const double UPPER_LOAD_FACTOR = 0.75;
+static const double LOWER_LOAD_FACTOR= 0.25;
+
+template <typename Value>
+class HashMap {
+private:
+  struct bucket {
+    std::string key;
+    Value value;
+    bucket() {}
+    bucket(const std::string& k, const Value& v): key(k), value(v) {}
+  };
+  bucket* buckets;
+  size_t len;
```

Comment:
I always like to add comments for what the variable is exactly for.
i.e. in case of multiple data structures -> what the len is a representation of